### PR TITLE
add more check on LSTMCell

### PR DESCRIPTION
**Pull request by fffrog**
Just like #107223, operator ``LSTMCell`` have the same problems as ``GRUCell``, and add some check and tests related to fix it.

### CODE REVIEW

File: torch/nn/modules/rnn.py [hunk 1]
```diff
@@ -1283,6 +1283,10 @@ def __init__(self, input_size: int, hidden_size: int, bias: bool = True,
     def forward(self, input: Tensor, hx: Optional[Tuple[Tensor, Tensor]] = None) -> Tuple[Tensor, Tensor]:
         if input.dim() not in (1, 2):
             raise ValueError(f"LSTMCell: Expected input to be 1D or 2D, got {input.dim()}D instead")
+        if hx is not None:
+            for idx, value in enumerate(hx):
+                if value.dim() not in (1, 2):
+                    raise ValueError(f"LSTMCell: Expected hidden{idx} to be 1D or 2D, got {value.dim()}D instead")
```
Comment (mikaylagawarecki) [inline torch/nn/modules/rnn.py:1289]: Hm for consistency with the documentation seems like there will be 2 Tensors in hx here and we call them `h_0` and `c_0` so would prefer not to call this `hidden{idx}`

Alternatively could we could we rephrase this to
```suggestion
                    raise ValueError(f"LSTMCell: Expected hx[{idx}] to be 1D or 2D, got {value.dim()}D instead")
```

Comment (FFFrog) [reply torch/nn/modules/rnn.py]: Thanks for your comment, fixed.

**Review by ezyang — APPROVED** [review]
